pkg/cloud/amazon/s3_storage.go

```diff
@@ -232,21 +274,26 @@ func parseS3URL(_ cloud.ExternalStorageURIContext, uri *url.URL) (cloudpb.Extern
 	}
 
 	conf.Provider = cloudpb.ExternalStorageProvider_s3
-	assumeRole, delegateRoles := cloud.ParseRoleString(s3URL.ConsumeParam(AssumeRoleParam))
+	assumeRoleValue := s3URL.ConsumeParam(AssumeRoleParam)
+	assumeRoleProvider, delegateRoleProviders := cloud.ParseRoleProvidersString(assumeRoleValue)
+	assumeRole, delegateRoles := cloud.ParseRoleString(assumeRoleValue)
```

Comment:
Can we add a comment here outlining the mixed version state we're covering for by continuing to write to RoleArn and DelegateRoleARNs? Also a TODO that we can remove this in 23.2?

Reply:
Done, added TODO.